symforce/slam/imu_preintegration/imu_factor.h

```diff
@@ -0,0 +1,192 @@
+/* ----------------------------------------------------------------------------
+ * SymForce - Copyright 2022, Skydio, Inc.
+ * This source code is under the Apache 2.0 license found in the LICENSE file.
+ * ---------------------------------------------------------------------------- */
+
+#pragma once
+
+#include <vector>
+
+#include <Eigen/Core>
+
+#include <sym/pose3.h>
+#include <symforce/opt/factor.h>
+#include <symforce/opt/key.h>
+
+#include "./preintegrated_imu_measurements.h"
+
+namespace sym {
+
+/**
+ * A factor for using on-manifold IMU preintegration in a SymForce optimization problem.
+ *
+ * Assumes IMU bias is constant over the preintegration window. Does not recompute the
+ * preintegrated measurements for the IMU bias arguments, but rather a first order
+ * approximation linearized at the the IMU biases given during preintegration.
```

Comment:
Maybe "Does not recompute the preintegrated measurements when the IMU bias estimate changes during optimization", also typo `the the`

Reply:
Thanks. That's clearer. Done.